### ISSUE TITLE
PUT request fails when both utf8 and binary form-data is sent

### ISSUE THREAD

**Issue by sunknudsen** [sindresorhus/got#1555]
#### Describe the bug

 - Node.js version: v12.19.0
 - OS & version: macOS Catalina 10.15.7

I am trying to update the title and thumbnail of a video on PeerTube using [form-data](https://www.npmjs.com/package/form-data).

The requests works fine as long as I don't send `thumbnailfile` which (I believe) [expects](https://docs.joinpeertube.org/api-rest-reference.html#tag/Video/paths/~1videos~1{id}/put) `binary` encoding.

I am using [get-stream](https://www.npmjs.com/package/get-stream) so the request can be [retried](https://github.com/sunknudsen/tube-manager/blob/master/src/peertube.ts#L22-L45) when client is not authenticated.

My (naive) gut feeling is that `get-stream` is converting `binary` encoding of `thumbnailfile` to `utf8`.

Perhaps I am missing something?

#### Actual behavior

PUT request fails

#### Expected behavior

PUT request succeeds

#### Code to reproduce

```typescript
import got from "got"
import formData from "form-data"
import getStream from "get-stream"
import { readFile } from "fs"
import { promisify } from "util"
import { parse } from "path"

const readFileAsync = promisify(readFile)

const thumbnail =
  "/Users/sunknudsen/Code/sunknudsen/sunknudsen-website/tube-manager/uJBgb8XJoA8.jpg"

!(async function () {
  const form = new formData()
  form.append("name", "foo")
  form.append(
    "thumbnailfile",
    await readFileAsync(thumbnail, { encoding: "binary" }),
    {
      contentType: "image/jpeg",
      filename: parse(thumbnail).base,
    }
  )
  await got.put(
    `https://peertube.sunknudsen.com/api/v1/videos/db71b05a-f648-4bdf-a037-6f91827ed0f3`,
    {
      headers: form.getHeaders(),
      body: await getStream(form),
    }
  )
})()
```

#### Checklist

- [ x ] I have read the documentation.
- [ x ] I have tried my code with the latest version of Node.js and Got.

**Comment by szmarczak** [comment]
```
/Users/sunknudsen/Code/sunknudsen/sunknudsen-website/tube-manager/uJBgb8XJoA8.jpg
```

Well, this path doesn't exist on my computer... Can you please post something actually reproducible?

**Comment by sunknudsen** [comment]
Hey @szmarczak, thanks for the follow-up.

Unfortunately, putting together a  reproducible example would require a lot of work given the PeerTube API requires authentication (I would have to build a mock API etc...).

I ended up dropping `get-stream` in favor of the following [approach](https://github.com/sindresorhus/got/discussions/1513#discussioncomment-187413) (using `genBody`) and it solved the issue.

My gut feeling is that `get-stream` (which detaults to `utf8`) was messing up the encoding of `thumbnailfile`.

That being said, it's likely I was messing something up at my end.

Closing the issue. Thanks for `got` btw! ❤️
